engine/docker/container.go

```diff
@@ -40,6 +46,34 @@ func (e *Engine) VirtualizationCreate(ctx context.Context, opts *enginetypes.Vir
 	if opts.Lambda {
 		opts.LogType = "json-file"
 	}
+	// set restart always
+	restartRetryCount := 3
+	if opts.RestartPolicy == restartAlways {
+		restartRetryCount = 0
+	}
+	// network mode 和 networks 互斥
+	// 没有 networks 的时候用 networkmode 的值
+	// 有 networks 的时候一律用用 networks 的值作为 mode
+	networkMode := dockercontainer.NetworkMode(opts.Network)
+	for name := range opts.Networks {
+		networkMode = dockercontainer.NetworkMode(opts.Network)
+		opts.Networks[name] = ""
```

Comment:
好像..... 会导致 replace 不能复用 ip.....

Reply:
> 不会…是cli参数问题吧 zc <notifications@github.com> 于 2019年10月18日周五 17:03写道：
> […](#)
> ***@***.**** commented on this pull request. ------------------------------ In engine/docker/container.go <[#134 (comment)](https://github.com/projecteru2/core/pull/134#discussion_r336389127)>: > @@ -40,6 +46,34 @@ func (e *Engine) VirtualizationCreate(ctx context.Context, opts *enginetypes.Vir if opts.Lambda { opts.LogType = "json-file" } + // set restart always + restartRetryCount := 3 + if opts.RestartPolicy == restartAlways { + restartRetryCount = 0 + } + // network mode 和 networks 互斥 + // 没有 networks 的时候用 networkmode 的值 + // 有 networks 的时候一律用用 networks 的值作为 mode + networkMode := dockercontainer.NetworkMode(opts.Network) + for name := range opts.Networks { + networkMode = dockercontainer.NetworkMode(opts.Network) + opts.Networks[name] = "" 好像..... 会导致 replace 不能复用 ip..... — You are receiving this because you modified the open/close state. Reply to this email directly, view it on GitHub <#134?email_source=notifications&email_token=AAD3S2A4D6RYM346ABT4GE3QPF3WXA5CNFSM4IVFMNM2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCINO6LY#pullrequestreview-303755055>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAD3S2ADENEE2L3SP6TBXMTQPF3WXANCNFSM4IVFMNMQ> .

应该不是... 下面调用 docker client API 的时候 networkConfig 里的 endpoint 里是空的, 因为这里把 `opts.Networks[name]` 里的 IP 置空了..